Backend/Records/Repository/MySqlRepository/MedicalRecordSqlRepository.cs

```diff
@@ -2,28 +2,45 @@
 using System.Collections.Generic;
 using System.Text;
 using Backend.Records.Model;
+using Model;
 using Model.Users;
 using Repository;
 using Repository.MedicalRecordRepository;
+using System.Linq;
 
 namespace Backend.Records.Repository.MySqlRepository
 {
-    class MedicalRecordSqlRepository : MySqlrepository<MedicalRecord, int>,
+    public class MedicalRecordSqlRepository : MySqlrepository<MedicalRecord, int>,
         IMedicalRecordRepository
     {
+        public MedicalRecordSqlRepository(MySqlContext context) : base(context) { }
+
         public IEnumerable<MedicalRecord> FilterRecordsByState(PatientCondition state)
         {
             throw new NotImplementedException();
         }
 
-        public MedicalRecord GetRecordBy(Patient patient)
+        public MedicalRecord GetMedicalRecordByPatientId(string id)
         {
-            throw new NotImplementedException();
+            List<MedicalRecord> medicalRecords = GetAll().ToList();
+            MedicalRecord medicalRecord = medicalRecords.FirstOrDefault(entity => entity.Patient.Id.Equals(id));
+
+            if (medicalRecord != null)
+            {
+                return medicalRecord;
+            }
+            else
+                throw new NotImplementedException();
```

Comment:
Else is unnecessary, if the if clause is true it will return the value.
And it should throw patient not found exception, not not implemented exception.